frontend/src/main-components/CreateCardContent.js

```diff
@@ -49,22 +48,13 @@ class CreateCardContent extends Component {
     // Ensures that languages have been selected before translating
     if (this.state.fromLang !== "none" && this.state.toLang !== "none") {
       (async () => {
-        try {
-          const translated = await getTranslation(
-            text,
-            this.state.fromLang,
-            this.state.toLang
-          );
-          if (!translated.ok) {
-            throw Error(translated.statusText);
-          }
-          this.setState({ translation: translated, text: text });
-        } catch (error) {
-          this.setState({
-            translation: "Text could not be translated",
-            text: text,
-          });
-        }
+        // If an error is thrown it is caught inside of get translation
+        const translated = await getTranslation(
```

Comment:
Need to cancel previous translation requests as well and avoid setting state. Otherwise, the user might see some UI jank if two requests fire in quick succession, and the second one returns before the first one.

1. fromLang = English
2. User sets toLang = Spanish (request gets fired)
3. User sets toLang = French (request gets fired)
4. French request comes back; translation is set to the French translation
5. Spanish request comes back; translation is set to the Spanish translation

Now the UI still shows toLang = French, but the translation is in Spanish.

One way to do this is to wrap setState in an if statement that checks if the current toLang is equal to the toLang when the translation was sent off:

```javascript
const toLang = this.state.toLang;
const translated = await getTranslation(...);
this.setState(prevState => {
  if (prevState.toLang != toLang || prevState.text != text) {
    return;
  }
  return { ...prevState, translation: translated.translation, text: text };
});
```

I also check text because in some circumstances (bad network connection) the original text may have changed before the translation request comes back.

Reply:
done.